vendor/k8s.io/kubernetes/cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -186,10 +186,10 @@ func Run(s *options.CMServer) error {
 
 		if StartInformers == nil {
 			ctx.InformerFactory.Start(ctx.Stop)
+			close(ctx.InformersStarted)
 		} else {
 			StartInformers(ctx.Stop)
 		}
-		close(ctx.InformersStarted)
```

Comment:
So As I understand it - the reason we had to make this change is because - in case of openshift (where `StartInformers` is not nil), we do not want to close the `ctx.InformersStarted` channel immediately after calling `StartInformers` because we want control over that in openshift code. 

Do we want to backport this to upstream at all?

Reply:
> So As I understand it - the reason we had to make this change is because - in case of openshift (where StartInformers is not nil), we do not want to close the ctx.InformersStarted channel immediately after calling StartInformers because we want control over that in openshift code.

Correct


> Do we want to backport this to upstream at all?

No.  In 3.9, we plan to start executing this separately.